enrollment-server/src/main/java/com/wultra/app/docverify/zenid/provider/ZenidDocumentVerificationProvider.java

```diff
@@ -323,6 +325,40 @@ public void cleanupDocuments(OwnerId id, List<String> uploadIds) throws Document
         return errors;
     }
 
+    @Override
+    public VerificationSdkInfo initVerificationSdk(OwnerId id, Map<String, String> initAttributes) throws DocumentVerificationException {
+        Preconditions.checkArgument(initAttributes.containsKey(ZenidConst.SDK_INIT_TOKEN), "Missing initialization token for ZenID SDK");
+        String token = initAttributes.get(ZenidConst.SDK_INIT_TOKEN);
+
+        ResponseEntity<ZenidWebInitSdkResponse> responseEntity;
+        try {
+            responseEntity = zenidApiService.initSdk(token);
+        } catch (RestClientException e) {
+            logger.warn("Failed REST call to init ZenID SDK, " + id, e);
+            throw new DocumentVerificationException("Unable to initialize the SDK due to a REST call failure");
+        } catch (Exception e) {
+            logger.error("Unexpected error when initializing ZenID SDK, " + id, e);
+            throw new DocumentVerificationException("Unexpected error when initializing ZenID SDK");
+        }
+
+        if (responseEntity.getBody() == null) {
+            logger.error("Missing response body when initializing ZenID SDK, " + id);
+            throw new DocumentVerificationException("Unexpected error when initializing ZenID SDK");
+        }
+
+        if (!HttpStatus.OK.equals(responseEntity.getStatusCode())) {
```

Comment:
What about to call `!responseEntity.getStatusCode().is2xxSuccessful()`

Reply:
Yeps, this is also a possible but it is much more wide variant. The REST API is returning 200, all other response codes I do consider as strange and weird.